app/rust/src/parser/transaction_payload.rs

```diff
@@ -64,27 +66,39 @@ impl<'a> StxTokenTransfer<'a> {
 
     pub fn memo(&self) -> &[u8] {
         let at = self.0.len() - 34;
-        &self.0[at..]
+        // safe to unwrap as parser checked for proper len
+        self.0.get(at..).apdu_unwrap()
     }
 
     pub fn amount(&self) -> Result<u64, ParserError> {
         let at = self.0.len() - 34 - 8;
-        be_u64::<'a, ParserError>(&self.0[at..])
-            .map(|res| res.1)
-            .map_err(|_| ParserError::parser_unexpected_buffer_end)
+        if let Some(amount) = self.0.get(at..) {
```

Comment:
this is more style.... match or use `.ok_or(no_data).and_then(|amount| be_u64(amount).map(...).map_err(...))?`